erpnext/payroll/doctype/salary_slip/salary_slip.js

```diff
@@ -214,14 +214,16 @@ frappe.ui.form.on('Salary Slip Timesheet', {
 });
 
 var calculate_totals = function(frm) {
-	if (frm.doc.earnings || frm.doc.deductions) {
-		frappe.call({
-			method: "set_totals",
-			doc: frm.doc,
-			callback: function() {
-				frm.refresh_fields();
-			}
-		});
+	if (frm.doc.docstatus === 0) {
+		if (frm.doc.earnings || frm.doc.deductions) {
+			frappe.call({
+				method: "set_totals",
+				doc: frm.doc,
+				callback: function() {
+					frm.refresh_fields();
+				}
+			});
+		}
```

Comment:
Looks fine. 
It also made me look at the refresh function. Unrelated but since this file is touched , the function `toggle_fields` in `refresh` can easily be moved inside the Doctype specification **Display Depends On**.